### PR TITLE
Fix cache restoration for elements with context

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium-uiautomator2-server/issues/280

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ protected AppiumResponse safeHandle(IHttpRequest request) throws UiObjectNotFoun
         }
 
         String id = UUID.randomUUID().toString();
-        AndroidElement androidElement = getAndroidElement(id, firstObject, null);
+        AndroidElement androidElement = getAndroidElement(id, firstObject, false);
```
Comment (KazuCocoa) [inline app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java]: Should not here be `true`?

File: app/src/main/java/io/appium/uiautomator2/model/KnownElements.java [hunk 2]
```diff
@@ -47,6 +47,67 @@ private String getCacheKey(AndroidElement element) {
...
     }
 
+    private void restoreCachedElement(AndroidElement element) {
+        final By by = element.getBy();
+        if (by == null) {
+            throw new StaleElementReferenceException(String.format(
+                    "The element '%s' does not exist in DOM anymore", element.getId()));
+        }
+
+        // We cannot restore a single element from a locator that matches multiple items
```
Comment (KazuCocoa) [inline app/src/main/java/io/appium/uiautomator2/model/KnownElements.java]: 👍

File: app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java (hunk 1 above)
Comment (KazuCocoa) [reply app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java]: ah... visible element might be multiple. we cannot say it is an exactly single element, so false...?

Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java]: exactly

**Pull request merged**